### PR TITLE
Change module .load and .conf files in /etc/apache2/mods-enabled into links

### PR DESCRIPTION
**Pull request by Aethylred**
fixes issue puppetlabs/puppetlabs-apache#160

### CODE REVIEW

File: manifests/mod.pp [hunk 1]
```diff
@@ -46,4 +46,33 @@
     require => Package['httpd'],
     notify  => Service['httpd'],
   }
+
+  if $::osfamily == 'Debian' {
+    $enable_dir = $apache::params::mod_enable_dir
+    file{ "enable.${mod}.load":
```
Comment (hunner) [inline manifests/mod.pp:52]: What is the reason for prefixing the `.load` files with `enable.` ? It looks like Debian and Ubuntu would normally just use `${mod}.load`

Comment (hunner) [inline manifests/mod.pp:52]: Oh, title conflicts for the symlink with the real load file. Got it.

**Comment by Aethylred** [comment]
Thinking about doing something similar for site/vhost files, but I don't need it right now.

**Comment by Aethylred** [comment]
It means that puppet creates and maintains the links and is aware of them, and can freely purge the rest of the `/etc/apache2mods-enabled` content. Purging `/etc/apache2/mods-available` directory would be awkward to work with.

**Pull request closed without merging**